src/Castle.Core.Tests/DynamicProxy.Tests/BasicClassProxyTestCase.cs

```diff
@@ -380,6 +380,21 @@ public void ClassProxyShouldCallProtectedDefaultConstructor()
 			Assert.AreEqual("Something", ((ClassWithProtectedDefaultConstructor)proxy2).SomeString);
 		}
 
+		[Test]
+		public void ClassProxyShouldHaveDefaultConstructorWhenBaseClassHasPrivateProtected()
+		{
+			object proxy = generator.CreateClassProxy<ClassWithPrivateProtectedConstructor>();
+			Assert.IsNotNull(Activator.CreateInstance(proxy.GetType()));
```

Comment:
I initially had a bit of trouble understanding this test. All I initially  saw was that you're creating two instances of the proxy type, which didn't make much sense until I realized the second instantiation is only a means to an end. I think this test would be easier to understand if it explicitly looked for the expected constructor using reflection, instead of assuming that `Activator` will do it for us.

Reply:
Honestly, I just copied both of these tests from the three other sets of nearly identical existing pairs of tests immediately above for different constructor accessibilities.

From the perspective of this specific fix, all that is strictly needed is one test which has one line of code which calls generator.CreateClassProxy<ClassWithPrivateProtectedConstructor>(), to verify that it doesn't throw. However, I thought that I should go with consistency with the existing tests, which is why you see these extra lines of code in the tests.
Considering that, would you still like me to make the change that you suggested to this test? If so, would you like me to update the similar tests immediately above in this file as well?

Reply:
Consistency with the existing tests sounds reasonable (I didn't notice that's how you did it, sorry), and good enough for me.